backend/internal/controller/adapter/shot_message.go

```diff
@@ -52,11 +52,17 @@ func UnpackShotMsg(message io.Reader) (model.Shot, error) {
 	}
 
 	return model.Shot{
-		Speed: params.Speed,
-		Team:  teamID,
+		Speed:      params.Speed,
+		Team:       teamID,
+		ShotAtGoal: checkIfShotAtGoal(params.EndArea, teamID),
 	}, nil
 }
 
+func checkIfShotAtGoal(areaID int, teamID int) bool {
```

Comment:
I'm not sure if tests for this method are necessary but I would suggest adding them.

Reply:
Tests have been added.